### PR TITLE
test: Add tests for dependent-field component

### PR DESCRIPTION
**Pull request by yksolanki9**
Clickup: https://app.clickup.com/t/85zry9tdr

Code coverage:
Screenshot 2023-04-27 at 11 53 51 AM

### CODE REVIEW

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,137 @@
...
+    expect(component.displayValue).toEqual(dependentFieldValue);
+  });
+
+  it('registerOnChange', fakeAsync(() => {
+    const callbackFn = jasmine.createSpy('callbackFn');
+    component.registerOnChange(callbackFn);
+    expect(component.onChangeCallback).toEqual(callbackFn);
+  }));
+
+  it('registerOnTouched', () => {
```
Comment (jayfyle) [inline src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts:132]: Why does this test case not have fakeAsync?

**Review by jayfyle — APPROVED** [review]
Looks good to me, overall all specs seem fine. Although, there are no DOM tests. Maybe further specs can be included that check whether interacting with elements trigger the intended methods.

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts:132]: This is a sync event, so we don't need it

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,138 @@
...
+    const callbackFn = jasmine.createSpy('callbackFn');
+    component.registerOnTouched(callbackFn);
+    expect(component.onTouchedCallback).toEqual(callbackFn);
```
Comment (Julias0) [inline src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts:136]: This assertion compares functions. We should test against the blur event and see if the spy got triggered or not

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,138 @@
...
+  it('writeValue(): should write the formControl value to select field', () => {
+    const dependentFieldValue = 'Dep Field 1';
+    component.writeValue(dependentFieldValue);
+    expect(component.displayValue).toEqual(dependentFieldValue);
+  });
+
+  it('registerOnChange', fakeAsync(() => {
+    const callbackFn = jasmine.createSpy('callbackFn');
+    component.registerOnChange(callbackFn);
+    expect(component.onChangeCallback).toEqual(callbackFn);
```
Comment (Julias0) [inline src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts:130]: Same as below, we should trigger an input event (value coming from modal) and see if the change callback spy got triggered or not

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts (hunk 3 above)
Comment (yksolanki9) [reply src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts]: Resolved

File: src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts (hunk 2 above)
Comment (yksolanki9) [reply src/app/shared/components/dependent-fields/dependent-field/dependent-field.component.spec.ts]: Resolved

**Review by Julias0 — APPROVED** [review]

**Pull request merged**